build.gradle

```diff
@@ -115,10 +119,9 @@ task insertVersion() {
 	}
 }
 
-def insertProjectVersion(String project, String abbreviation) {
-	Properties properties = new Properties()
-	file("${project}/version.properties").withInputStream { properties.load(it) }
-	String version = properties.previousVersion
+def insertProjectVersion(final String project, String abbreviation) {
+	final def tags = grgit.open(dir: project).tag.list().collect { it.getName() }
+	final def version = tags.max()
```

Comment:
Three notes:

1. Below we are explicit about the types (`File`, `String`, …), here you use `def` (and `final`). Personally, I would prefer a consistent style.
2. I like the "one op per line" style:

```groovy
final def tags = grgit.open(dir: project)
        .tag
        .list()
        .collect { it.getName() }
```
3. Not that familiar with Gradle-Groovy, but would a method reference also work for `{ it.getName() }`?

Reply:
1. Yes
2. Yes
3. that code was removed